### DISCUSSION TITLE
Using Conventional Unit Cell in Anphon Calculation

### DISCUSSION THREAD

**Discussion by zohansyahfatomi** [ttadano/alamode#204] [Q&A]
Dear Admin,

My name is Zohan.
I am a Graduate Student at Kanazawa University.
I can not find the primitive cell in my structure.
My structure is SnSe (8 atoms: 4 Se and 4 Sn).

Sn4 Se4
1.0
   4.1262509500000002    0.0000000000000000    0.0000000000000003
   0.0000000000000008    4.7009074499999999    0.0000000000000003
   0.0000000000000000    0.0000000000000000   11.6644585000000003
Sn Se
4 4
direct
   0.7500000000000000    0.3765543000000000    0.6187868500000000 Sn2+
   0.2500000000000000    0.8765543000000000    0.8812131500000000 Sn2+
   0.2500000000000000    0.6234457000000000    0.3812131500000000 Sn2+
   0.7500000000000000    0.1234457000000000    0.1187868500000000 Sn2+
   0.2500000000000000    0.5091818000000000    0.1485530300000000 Se2-
   0.2500000000000000    0.9908182000000000    0.6485530300000000 Se2-
   0.7500000000000000    0.0091817999999999    0.3514469700000000 Se2-
   0.7500000000000000    0.4908182000000000    0.8514469700000000 Se2-

Is it possible to use a conventional unit cell for anphon calculation?
Thank you very much.

Sincerely yours,
Zohan

**Comment by ttadano** [comment]
In version 1.x.y of ALAMODE, using non-primitive cell is not supported because the anphon code internally detects a primitive cell and use it for further calculations.

In version 2.x (currently developed under 2.0dev branch), it is possible to use a non-primitive cell . If you are interested, please try 2.0dev.

**Reply by xinyu1905** [reply]
Thank you for the reply. i have tried 2.0dev and set the PRIMCELL parameter (use conventional cell), however, the forceconstants are still related to the true primitive cell，how should I get the forceconstants which are related to the non-primitive cell?

**Reply by ttadano** [reply]
The original question asks how to calculate phonons using a non primitive cell. This can be achieved by setting the non primitive cell parameters for the cell field of anphon. 

Do you ask the same question?

**Reply by xinyu1905** [reply]
I am sorry, my question differs from the original one. I would like to know how to obtain the force constants for a non-primitive cell. When ALAMODE calculates the force constants, it converts the non-primitive cell into a primitive cell. For instance, if the conventional cell contains 10 atoms, the primitive cell contains 5 atoms, and the supercell contains 60 atoms, ALAMODE provides force constants in a 5 x 60 format. However, I need the force constants in a 10 x 60 format. How should I proceed?

**Reply by ttadano** [reply]
I understand your question. 

Unfortunately, ALAMODE does not have the option to print the force constants in the format you requested. However, it is easy to convert the 5x60 format into 10x60 since you just need to translate all atoms by the lattice translation vector.

**Reply by xinyu1905** [reply]
Thank you very much, I have successfully completed the conversion. By the way, since the higher-order force constants are also based on the primitive cell, if I want to use the 2nd, 3rd, and 4th-order force constants in ShengBTE calculations, the 2nd-order force constants need to be adapted to the supercell. For example, if I need to convert the 5 x 60 force constants to 40 x 40 force constants (with the primitive cell expanded to 2 x 2 x 2), how should I proceed?

**Reply by ttadano** [reply]
I'm not sure if I understand your question, but basic stragegy remains the same; you just need to translate all atom triples (ijk) or quartets (ijkl) by the lattice translation vectors that span the supercell.

**Reply by xinyu1905** [reply]
Thank you for your suggestion, I will try it.

**Comment by zohansyahfatomi** [comment]
Dear Prof. Tadano,

Thank you for the reply.
I plan to download the version of 2.x, however in https://sourceforge.net/projects/alamode/, I think it is version 1.x. How to get the version of 2.x?
Thank you very much.

Sincerely yours,
Zohan

**Reply by gangliu-github** [reply]
U can find 2.x in https://github.com/ttadano/alamode/branches
